### PR TITLE
fix: create recursive_from_prefix path if it does not exist

### PR DESCRIPTION
**Pull request by Bento007**
# Reason
- #887 

# Changes
- create the recursive_from_prefix directory if it does not exist.
- add docstring to `get_destination_path`
- add unit tests for `get_destination_path`

### CODE REVIEW

File: client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py [hunk 1]
```diff
@@ -57,12 +57,29 @@ def get_destination_path(
...
+
+    """
     if not dest_path:
         dest_path = os.getcwd()
     dest_path = os.path.abspath(dest_path)
 
     if not os.path.isdir(dest_path) and recursive_from_prefix:
-        raise ValueError("Recursive downloads require a base directory")
+        raise ValueError(
+            f"The destination path '{dest_path}' does not exist. Recursive downloads require an existing base directory.",
```
Comment (Bento007) [inline client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:81]: @kandarpksk what are your thoughts on the modified error message.

**Review by manasaV3 — APPROVED** [review]
LGTM

File: client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py (hunk 1 above)
Comment (kandarpksk) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:81]: I just read recent comments in the [issue](https://github.com/chanzuckerberg/cryoet-data-portal/issues/887), and think we should just create the missing folder in all cases.

If we don't go with that, I find the updated message still confusing. I think the language we use should be from the perspective of the API user, who is either downloading a file (with say, download_mrcfile) or a folder (with say, download_everything). In which case is the download recursive? (I'd ordinarily assume that the folder download is recursive, but from [Manasa's comment](https://github.com/chanzuckerberg/cryoet-data-portal/issues/887#issuecomment-2251089496), it seems to me that in that case, this error is not thrown!)

File: client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py [hunk 2]
```diff
@@ -54,15 +54,27 @@ def download_https(
...
+
+    """
     if not dest_path:
         dest_path = os.getcwd()
     dest_path = os.path.abspath(dest_path)
 
-    if not os.path.isdir(dest_path) and recursive_from_prefix:
-        raise ValueError("Recursive downloads require a base directory")
+    if not os.path.isdir(dest_path):
+        os.makedirs(dest_path)
```
Comment (manasaV3) [inline client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:77]: When the dest_path is None, it should point to the working directory, and will pass this check. So, the only time this exception would show up would be when the user explicitly provides an invalid value and we should error on it. I think the error message could be more verbose in explaining the issue.  cc: @kandarpksk 

Also, [this line](https://github.com/chanzuckerberg/cryoet-data-portal/pull/940/files#diff-63d5c8af96144c2f98bba0f8110bc718612219f91e9b97e9b40e3d74a9a9bf73R85) should already be creating the recursive directory structure.

**Review by manasaV3 — CHANGES_REQUESTED** [review]

File: client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py [hunk 3]
```diff
@@ -54,15 +54,27 @@ def download_https(
 
 def get_destination_path(
     url: str,
-    dest_path: Optional[str],
+    dest_path: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (Bento007) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:57]: > Is there a case, where we are calling this method without a value for dest_path?

It's only used in [one](https://github.com/chanzuckerberg/cryoet-data-portal/blob/e9046e43cb7580daaefa7eb329b82aecb208f805/client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py#L118-L118) place, and dest_path is provided.

I see, you're using Optional to mean String or None while at the same time enforcing that dest_path must be a parameter.

File: client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py (hunk 2 above)
Comment (Bento007) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py:77]: So error when an invalid name is provided? Invalid meaning illegal path characters or maybe a path we can't write to?

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**